sources/Interop/Windows/shared/uuids/Windows.Manual.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+// Ported from shared/uuids.h in the Windows SDK for Windows 10.0.19041.0
+// Original source is Copyright © Microsoft. All rights reserved.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics.Contracts;
+using System.Runtime.CompilerServices;
+
+namespace TerraFX.Interop
+{
+    public static partial class Windows
+    {
+        /// <summary>
+        /// Retrieves the GUID of of a specified type.
+        /// </summary>
+        /// <typeparam name="T">The type to retrieve the GUID for.</typeparam>
+        /// <returns>
+        /// A <see cref="GUID"/> value wrapping a pointer to the GUID data for the
+        /// input type. This value can be either converted to a <see cref="Guid"/>
+        /// pointer, or implicitly assigned to a <see cref="Guid"/> value.
+        /// </returns>
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe GUID __uuidof<T>()
+            where T : unmanaged
+        {
+            return new(UUID<T>.RIID);
+        }
+
+        /// <summary>
+        /// A proxy type that wraps a pointer to GUID data. Values of this type can be implicitly
+        /// converted to and assigned to <see cref="Guid"/>* or <see cref="Guid"/> parameters.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public readonly unsafe ref struct GUID
```

Comment:
I don't like calling this `GUID`. `GUID` is an existing native type and so I'd rather give this a more descriptive name that identifies its a helper. Maybe `UuidOfType` or similar.

Reply:
Sounds good! Done in https://github.com/terrafx/terrafx.interop.windows/pull/151/commits/8702e3d0f93e5dbd69df9ff7c7acc64b63fd1f22 👍